### PR TITLE
sensor => scanner v4 api call

### PR DESCRIPTION
**Pull request by daynewlee**
## Description

Added a v4 grpc client as an implementation of interface client.
v4 grpc client calls scanner v4 from sensor and get the index report.
We need to utilize central client and send scanner v4 index report to scanner v4 matcher

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: sensor/common/scannerclient/grpc_client.go [hunk 1]
```diff
@@ -49,19 +56,19 @@ func dial(endpoint string) (*Client, error) {
 	// then add the grpc.WithBlock() DialOption.
 	conn, err := grpc.Dial(endpoint, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to dial Scanner")
+		return nil, errors.Wrap(err, "failed to Dial Scanner")
```
Comment (jvdm) [inline sensor/common/scannerclient/grpc_client.go:59]: `s/Dial/dial/`

File: sensor/common/scannerclient/v4_grpc_client.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	//return resp or return indexReport directly?
+	if err != nil {
+		log.Debugf("Failed to convert indexer report to image components from local Scanner for image %s: %v", name, err)
+		return nil, errors.Wrap(err, "converting indexer report from scanner")
+	}
+
+	return resp, nil
+}
+
+func convertIndexReportToV1GetImageComponentsResponse(indexReport scannerV4.IndexReport, image *storage.Image) (*scannerV1.GetImageComponentsResponse, error) {
```
Comment (jvdm) [inline sensor/common/scannerclient/v4_grpc_client.go:95]: Have you made sure that ImageComponents has all the necessary components for the Central's Scanner V4 Indexer and Matcher to be able to match vulnerabilities? 

Additionally, we need to make decisions about how the Matcher will retrieve the IndexReport to match vulnerabilities. I'm curious to know if we plan to store it in the database or somewhere else for the Matcher to access. See https://issues.redhat.com/browse/ROX-16919

**Review by jvdm — CHANGES_REQUESTED** [review]
Could you confirm if Scanner V4 has the capability to match vulnerabilities over a `scannerV1.GetImageComponentsResponse` or if we require more information? It's possible that we may need to expand the proto to incorporate the `IndexReport`, or augment it with extra fields if we opt to "unwrap" the index report. It is worth talking about the pros/cons a little bit to make sure.

File: sensor/common/scannerclient/v4_grpc_client.go (hunk 2 above)
Comment (daynewlee) [reply sensor/common/scannerclient/v4_grpc_client.go:95]: The second point of view makes sense to me. If we are going to save index report to DB for matcher to fetch them later, do we still have to worry about the first question?

**Pull request closed without merging**